### ISSUE TITLE
[EKS] [bug]: Fix kube-dns service port mapping for metrics

### ISSUE THREAD

**Issue by DaazKu** [aws/containers-roadmap#2177]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Want https://github.com/aws/containers-roadmap/issues/965 to be addressed. It's an actual bug.

**Which service(s) is this request for?**
EKS - CoreDNS addon

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
The core DNS service is defining itself as being scrapable by Prometheus on port 9153 but the port is not mapped which cause the scrapes to timeout.

**Are you currently working around this issue?**
We can't work around the issue because we do not have access to modify the service definition.
Any "hack patch" would be overridden when updating the addon.

Can be verified using:
```bash
aws eks describe-addon-configuration \
--addon-name coredns \
--addon-version v1.10.1-eksbuild.4 \
--query configurationSchema \
--output text | jq
```

**Additional context**
The fix is simply to add the missing port in the ports section of the service:

```yaml
kind: Service
apiVersion: v1
metadata:
  name: kube-dns
    
    # omitted information here
    
    prometheus.io/port: '9153'
    prometheus.io/scrape: 'true'
spec:
  ports:
    - name: dns
      protocol: UDP
      port: 53
      targetPort: 53
    - name: dns-tcp
      protocol: TCP
      port: 53
      targetPort: 53
    ############## WHAT NEEDS TO BE ADDED ###############
    - name: metrics
      protocol: TCP
      port: 9153
      targetPort: 9153
    ###################################################
# omitted information here
```

The containers are already setup to reply on that port:
```yaml
# omitted information here
  containers:
    - args:
        - -conf
        - /etc/coredns/Corefile
      image: 602401143452.dkr.ecr.ca-central-1.amazonaws.com/eks/coredns:v1.10.1-eksbuild.4
      imagePullPolicy: IfNotPresent
# omitted information here
      ports:
        - containerPort: 53
          name: dns
          protocol: UDP
        - containerPort: 53
          name: dns-tcp
          protocol: TCP
        - containerPort: 9153
          name: metrics
          protocol: TCP
  # omitted information here
  ```

**Comment by ChrisMcKee** [comment]
Quick and dirty fix (as we're patching)
Patching the service and adding a pod monitor

```shell
kubectl -n kube-system patch svc kube-dns -p '{"metadata": { "annotations": {"prometheus.io/port": "9153","prometheus.io/scrape": "true"}}, "spec": {"ports": [{"port": 9153,"targetPort": 9153,"name": "metrics"}]}}'

kubectl apply -f - <<EOF
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: kube-dns
  namespace: kube-system
  labels:
    k8s-app: kube-dns
    prometheus-visibility.monitoring: "enabled"
spec:
  jobLabel: app.kubernetes.io/name
  namespaceSelector:
    matchNames:
      - kube-system
  podMetricsEndpoints:
    - honorLabels: false
      honorTimestamps: true
      interval: 30s
      path: /metrics
      port: metrics
  selector:
    matchLabels:
      k8s-app: kube-dns
EOF
```

**Comment by sjastis** [comment]
This is now fixed in coredns v1.10.1-eksbuild.5, v1.9.3-eksbuild.9 and v1.8.7-eksbuild.8 that released last week. Let us know if you are still running into an issue.
